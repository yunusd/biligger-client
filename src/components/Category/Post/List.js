import React from 'react';
import { Query } from 'react-apollo';
import { Dimmer, Loader } from 'semantic-ui-react';

import Summary from './Summary';
import GET_LATEST_POSTS_BY_CATEGORY from './queries';

const List = (props) => {
  const path = props.location.pathname.slice(1);
  const name = path.replace('-', ' ');

  return (
    <Query query={GET_LATEST_POSTS_BY_CATEGORY} variables={{ category: name }}>
      {({ loading, error, data }) => {
        if (loading) {
          return (
            <Dimmer active inverted>
              <Loader inverted content="YÃ¼kleniyor" />
            </Dimmer>
          );
        }

        const isExist = data.getLatestPostsByCategory.length !== 0;

        return <Summary isExist={isExist} error={error} category={name} data={data} />;
      }}
    </Query>
  );
};

export default List;
