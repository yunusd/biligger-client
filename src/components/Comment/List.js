import React from 'react';
import { useQuery, useApolloClient } from 'react-apollo-hooks';
import { Link } from 'react-router-dom';
import moment from 'moment';
import removeMd from 'remove-markdown';

import {
 Comment, Card, Divider, Icon, Button,
} from 'semantic-ui-react';

import Like from '../Like';
import { GET_ME_FROM_CACHE, GET_AUTH_STATUS } from '../../queries';
import { GET_LATEST_COMMENTS } from './queries';
import dateLocale from '../../helpers/dateLocale';
import urlSerializer from '../../helpers/urlSerializer';
import './Comment.css';

moment.updateLocale('en', dateLocale);

const Reply = ({ parent, getMe, currentUser }) => {
  const { data, loading, error } = useQuery(GET_LATEST_COMMENTS, {
    variables: {
      parent: parent.id,
      offset: 0,
      limit: 1,
    },
    fetchPolicy: 'network-only',
  });
  if (loading) return null;
  if (error) return 'Biraz sıkıntı yaşıyoruz!';
  if (!data.getLatestComments.length) return null;
  const {
    id,
    author,
    content,
    like,
    createdAt,
  } = data.getLatestComments[0];

  const slug = urlSerializer({
    id,
    username: author.username,
    text: {
      content,
    },
    type: {
      comment: true,
    },
  });

  const auth = {
    isOwn: getMe && getMe.username === author.username,
    isLoggedIn: currentUser.isLoggedIn && true,
  };

  const raw = removeMd(content.replace(/\\/g, ''));

  const plainContent = raw.length < 500 ? raw : `${raw.slice(0, 500)}...`;

  return (
    <React.Fragment>
      <Divider clearing />
      <Comment>
        <Comment.Content>
          <Comment.Author className="comment-list-author" as={Link} to={`/@${author.username}`}>
            { author.username }
          </Comment.Author>
          <Comment.Metadata className="comment-list-meta">
            <span>{moment(createdAt).fromNow()}</span>
          </Comment.Metadata>
          <br />
          <Comment.Text
            className="comment-list-text"
            as={Link}
            to={`${slug.comment.url}`}
            style={{ fontSize: '18px', lineHeight: '1.618em' }}
          >
            { plainContent }
          </Comment.Text>
          <Comment.Actions className="comment-list-actions">
            <Like parentModel="Comment" id={id} like={like} />
            {auth.isLoggedIn && (
              auth.isOwn ? (
                <React.Fragment>
                  <Link to={`${slug.comment.url}/duzenle`} className="summary-context-right summary-context-icon">
                    <Icon name="edit" size="small" />
                  </Link>
                  {/* <DeletePost id={id} authorId={author.id} {...props} /> */}
                </React.Fragment>
              ) : (
                <Link to="#" title="Bildir" className="summary-context-right summary-context-icon">
                  <Icon name="flag" title="bildir" size="small" />
                </Link>
              )
            )}
          </Comment.Actions>
        </Comment.Content>
      </Comment>
    </React.Fragment>
  );
};

const CommentList = ({ data }) => {
  const client = useApolloClient();
  const { currentUser } = client.readQuery({ query: GET_AUTH_STATUS });
  const { getMe } = currentUser.isLoggedIn ? client.readQuery({ query: GET_ME_FROM_CACHE }) : false;


  return data.getLatestComments.map(({
    id,
    author,
    content,
    like,
    createdAt,
  }) => {
    const auth = {
      isOwn: getMe && getMe.username === author.username,
      isLoggedIn: currentUser.isLoggedIn && true,
    };

    const raw = removeMd(content.replace(/\\/g, ''));

    const slug = urlSerializer({
      id,
      username: author.username,
      text: {
        content: raw,
      },
      type: {
        comment: true,
      },
    });

    const plainContent = raw.length < 500 ? raw : `${raw.slice(0, 500)}...`;

    return (
      <Card fluid className="comment-list-card" key={id}>
        <Comment.Group className="comment-list-group">
          <Comment>
            <Comment.Content>
              <Comment.Author className="comment-list-author" as={Link} to={`/@${author.username}`}>
                { author.username }
              </Comment.Author>
              <Comment.Metadata className="comment-list-meta">
                <span>{moment(createdAt).fromNow()}</span>
              </Comment.Metadata>
              <br />
              <Comment.Text
                className="comment-list-text"
                as={Link}
                to={`${slug.comment.url}`}
                style={{ fontSize: '18px', lineHeight: '1.618em' }}
              >
                { plainContent }
              </Comment.Text>
              <Comment.Actions className="comment-list-actions">
                <Like parentModel="Comment" id={id} like={like} onList />

                {auth.isLoggedIn && (
                  auth.isOwn ? (
                    <React.Fragment>
                      <Link to={`${slug.comment.url}/duzenle`} className="summary-context-right summary-context-icon">
                        <Icon name="edit" size="small" />
                      </Link>
                    </React.Fragment>
                  ) : (
                    <Link to="#" title="Bildir" className="summary-context-right summary-context-icon">
                      <Icon name="flag" title="bildir" size="small" />
                    </Link>
                  )
                )}
              </Comment.Actions>
            </Comment.Content>
          </Comment>
          <Reply parent={{ id }} getMe={getMe} currentUser={currentUser} />
        </Comment.Group>
      </Card>
    );
  });
};

const List = ({ parent }) => {
  const {
 data, loading, error, fetchMore,
} = useQuery(GET_LATEST_COMMENTS, {
    variables: {
      parent: parent.id,
      offset: 0,
      limit: 10,
    },
    fetchPolicy: 'network-only',
  });
  if (loading) return null;
  if (error) return 'Biraz sıkıntı yaşıyoruz!';
  if (!data.getLatestComments.length) return 'Yorum Yok!';

  return (
    <React.Fragment>
      <CommentList data={data} />
      {data.getLatestComments.length >= 10
      && (
        <Button
          basic
          fluid
          onClick={() => {
            fetchMore({
              variables: {
                offset: data.getLatestComments.length,
              },
              updateQuery: (prev, { fetchMoreResult }) => {
                if (!fetchMoreResult) return prev;
                return Object.assign({}, prev, {
                  getLatestComments: [...prev.getLatestComments, ...fetchMoreResult.getLatestComments],
                });
              },
            });
          }}
        >
          Daha fazla
        </Button>
      )}
    </React.Fragment>
  );
};

export default List;
