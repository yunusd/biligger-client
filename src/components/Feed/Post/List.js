import React from 'react';
import { Query } from 'react-apollo';
import {
 Dimmer, Loader, Button,
} from 'semantic-ui-react';
import { useApolloClient } from 'react-apollo-hooks';

import { GET_POPULER_POSTS } from './queries';
import { GET_ME_FROM_CACHE, GET_AUTH_STATUS } from '../../../queries';

import Summary from './Summary';

const List = () => {
  const client = useApolloClient();

  const { currentUser } = client.readQuery({ query: GET_AUTH_STATUS });
  const { getMe } = currentUser.isLoggedIn ? client.readQuery({ query: GET_ME_FROM_CACHE }) : false;

  return (
    <Query
      query={GET_POPULER_POSTS}
      variables={{
        offset: 0,
        limit: 10,
      }}
    >
      {({
        loading, error, data, fetchMore,
      }) => {
        if (loading) {
          return (
            <Dimmer active inverted>
              <Loader inverted content="Yükleniyor" />
            </Dimmer>
          );
        }

        const isExist = data.getPopulerPosts.length !== 0;

        if (!isExist) {
          return (
            <h5>Bilig bulunamadı!</h5>
          );
        }
        return (
          <React.Fragment>
            { error && 'Maalesef bazı sıkıntılar yaşıyoruz!'}
            <Summary data={data.getPopulerPosts} getMe={getMe} currentUser={currentUser} />
            {data.getPopulerPosts.length >= 10
              && (
                <Button
                  basic
                  fluid
                  onClick={() => {
                    fetchMore({
                      variables: {
                        offset: data.getPopulerPosts.length,
                      },
                      updateQuery: (prev, { fetchMoreResult }) => {
                        if (!fetchMoreResult) return prev;
                        return Object.assign({}, prev, {
                          getPopulerPosts: [...prev.getPopulerPosts, ...fetchMoreResult.getPopulerPosts],
                        });
                      },
                    });
                  }}
                >
                  Daha Fazla
                </Button>
            )}
          </React.Fragment>
        );
      }}
    </Query>
  );
};

export default List;
