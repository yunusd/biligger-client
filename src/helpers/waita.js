export default (amount = 0) => new Promise(resolve => setTimeout(resolve, amount));
