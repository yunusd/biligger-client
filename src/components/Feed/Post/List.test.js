import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

import { MockedProvider } from 'react-apollo/test-utils';
import { mount, shallow } from 'enzyme';

import waita from '../../../helpers/waita';

import GET_LATEST_POSTS from './queries';
import List from './List';

describe('List', () => {
  const mock = [
    {
      request: {
        query: GET_LATEST_POSTS,
      },
      result: {
        data: {
          getLatestPosts: [
            {
              id: 1,
              title: 'Title',
              content: 'Content',
              url: 'https://google.com',
              createdAt: Date.now(),
              author: {
                username: 'Foo',
              },
            },
          ],
        },
      },
    },
  ];

  it('render without crashing', async () => {
    const wrapper = mount(
      <MockedProvider mocks={mock} addTypename={false}>
        <Router>
          <List />
        </Router>
      </MockedProvider>,
    );
    await waita(1);
    const output = !(wrapper.html().search('Foo') < 0); // if can't find anything then returning -1
    expect(output).toBeTruthy();
  });

  it('render correctly', () => {
    const wrapper = shallow(<List />);
    expect(wrapper).toMatchSnapshot();
  });
});
