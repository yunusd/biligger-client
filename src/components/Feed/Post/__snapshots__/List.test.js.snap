// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
      "query": Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": undefined,
            "operation": "query",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "getLatestPosts",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "title",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "content",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "url",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "author",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "username",
                              },
                              "selectionSet": undefined,
                            },
                          ],
                        },
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "createdAt",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
            },
            "variableDefinitions": Array [],
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 137,
          "start": 0,
        },
      },
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
        "query": Object {
          "definitions": Array [
            Object {
              "directives": Array [],
              "kind": "OperationDefinition",
              "name": undefined,
              "operation": "query",
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "alias": undefined,
                    "arguments": Array [],
                    "directives": Array [],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "getLatestPosts",
                    },
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "id",
                          },
                          "selectionSet": undefined,
                        },
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "title",
                          },
                          "selectionSet": undefined,
                        },
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "content",
                          },
                          "selectionSet": undefined,
                        },
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "url",
                          },
                          "selectionSet": undefined,
                        },
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "author",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "username",
                                },
                                "selectionSet": undefined,
                              },
                            ],
                          },
                        },
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "createdAt",
                          },
                          "selectionSet": undefined,
                        },
                      ],
                    },
                  },
                ],
              },
              "variableDefinitions": Array [],
            },
          ],
          "kind": "Document",
          "loc": Object {
            "end": 137,
            "start": 0,
          },
        },
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
